### DISCUSSION TITLE
Preserves reference equality for requests

### DISCUSSION THREAD

**Discussion by codingnuclei** [urql-graphql/urql#1661] [Q&A]
Hi,

I'm looking at using urql core within angular and have been looking at the bindings for inspiration.

In react I stumbled across this:

https://github.com/FormidableLabs/urql/blob/720f60c6b118b80dc2bef2fc0a0b22aab23e5415/packages/react-urql/src/hooks/useRequest.ts#L5-L22

Just wanting to ask the question about why preserving reference equality for requests is important? What does it mean in terms of urql.

Thanks in advance :)

**Comment by JoviDeCroock** [comment]
It's pretty ergonomic when it comes to React-hooks as you can use them as a hook-dependency and don't have to fear referential equality making a hook rerun.

**Reply by codingnuclei** [reply]
Ah ok cool. Makes sense. 

Thanks!
